src/Parser/TemplateParser.cs

```diff
@@ -138,20 +151,78 @@ private string ParseName(out int parameterIndex)
                 c = Peek();
                 if (c == '}' || c == ':' || c == ',')
                     parameterIndex = parsed;
-                _position = start;        
+                _position = start;
             }
 
             if (parameterIndex == -1)
                 _isPositional = false;
 
-            return ReadUntil(HoleDelimiters);            
+            return ReadUntil(HoleDelimiters);
         }
 
+        /// <summary>
+        /// Parse format after hole name/index
+        /// </summary>
+        /// <returns></returns>
         private string ParseFormat()
         {
             Skip(':');
-            // TODO: Escaped }} in formats?
-            return ReadUntil('}');
+            var format = ParseFormatInner();
```

Comment:
I would turn recursion into a loop here.
It would probably be less code and more efficient.